src/app/application.cpp

```diff
@@ -335,7 +335,10 @@ void Application::runExternalProgram(const BitTorrent::TorrentHandle *torrent) c
 
     ::LocalFree(args);
 #else
-    QProcess::startDetached(QLatin1String("/bin/sh"), {QLatin1String("-c"), program});
+    // Cannot give users shell environment by default, as doing so
+    // might enable command injection when some automation has been setup.
```

Comment:
I would modify this message to focus more on the source of injection

```
Cannot give users shell environment by default, as doing so could
enable command injection via torrent name and other arguments

Reply:
sure, updated.